plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -138,9 +142,25 @@ protected void copyTableSchema(Connection connection, String catalogName, String
     protected String createTableSql(RemoteTableName remoteTableName, List<String> columns, ConnectorTableMetadata tableMetadata)
     {
         ImmutableList.Builder<String> tableOptions = ImmutableList.builder();
-        ClickHouseTableProperties.getEngine(tableMetadata.getProperties())
-                .ifPresent(value -> tableOptions.add("ENGINE =" + quoted(value)));
-        return format("CREATE TABLE %s (%s) %s", quoted(remoteTableName), join(", ", columns), join(", ", tableOptions.build()));
+        Map<String, Object> tableProperties = tableMetadata.getProperties();
+        ClickHouseEngineType engine = ClickHouseTableProperties.getEngine(tableProperties);
+        tableOptions.add("ENGINE = " + engine.getEngineType());
+        log.info("engine = " + engine);
+        if (engine == ClickHouseEngineType.MERGETREE) {
+            // order_by property is required
+            if (ClickHouseTableProperties.getOrderBy(tableProperties).isEmpty()) {
+                throw new TrinoException(INVALID_TABLE_PROPERTY, format("%s is required for %s", ClickHouseTableProperties.ORDER_BY_PROPERTY, engine.getEngineType()));
+            }
+            else {
+                ClickHouseTableProperties.getOrderBy(tableProperties).ifPresent(value -> tableOptions.add("ORDER BY " + value));
+            }
+            ClickHouseTableProperties.getPrimaryKey(tableProperties).ifPresent(value -> tableOptions.add("PRIMARY KEY " + value));
```

Comment:
Does `value` need to be quoted?

Reply:
Actually, I'm not quite sure what the difference is between adding the `quoted` function or not, is it to escape special characters or to prevent injection?

Reply:
I tested, `quoted` function will add double quotes to value, It's wrong for create table statement.